docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+
+```cpp
+ACTION hello::addmsg(name nm, std::string message) {
+
+    check(has_auth(user), "User is not authorized to perform this action.");
+
+    messages_table messages(get_self(), get_self().value);
+
+    messages.emplace(get_self(), [&](auto& row) {
+        row.user = nm;
+        row.message = message;
+        row.messagecks = eosio::sha256(message.data(), message.size());
+        row.time = current_time_point().time_since_epoch().count();
+    });
+}
+```
+
+If you compile and deploy the smart contract and send the `addmsg` action with first parameter `ama` and sign it with `hello@active` key it will fail with the custom error message defined:
+
+```cpp
+dune --send-action hello addmsg '[ama, "hi again"]' hello@active
+```
+
+```txt
+failed transaction: 6d2e11e24d9adc066136f94bc66c13df2dfce952f1a5a0fa7a0286043a67f0c6  <unknown> bytes  <unknown> us
+error 2023-03-01T15:44:02.588 cleos     main.cpp:700                  print_result         ] soft_except->to_detail_string(): 3050003 eosio_assert_message_exception: eosio_assert_message assertion failure
+assertion failure with message: User is not authorized to perform this action.
+    {"s":"User is not authorized to perform this action."}
+    nodeos  cf_system.cpp:14 eosio_assert
+pending console output: 
+    {"console":""}
+    nodeos  apply_context.cpp:124 exec_one
+```
+
+### Use require_auth()
+
+It does the same thing as the previous combination only that you can not customize the error message raised in case of failure.
+
+```cpp
+ACTION hello::addmsg(name nm, std::string message) {
+
+    require_auth( nm );
+
+    messages_table messages(get_self(), get_self().value);
+
+    messages.emplace(get_self(), [&](auto& row) {
+        row.user = nm;
+        row.message = message;
+        row.messagecks = eosio::sha256(message.data(), message.size());
+        row.time = current_time_point().time_since_epoch().count();
+    });
+}
+```
+
+If you compile and deploy the smart contract and send the `addmsg` action with first parameter `ama` and sign it with `hello@active` key it will fail with a standard error message:
+
+```cpp
+dune --send-action hello addmsg '[ama, "hi again"]' hello@active
+```
+
+```txt
+failed transaction: 8887cca7fababeb883aa6806c220eece3d5f3c618824b2378ac25a67c09a063a  <unknown> bytes  <unknown> us
+error 2023-03-01T15:53:35.033 cleos     main.cpp:700                  print_result         ] soft_except->to_detail_string(): 3090004 missing_auth_exception: Missing required authority
+missing authority of ama
+    {"account":"ama"}
+    nodeos  apply_context.cpp:256 require_authorization
+pending console output: 
+    {"console":""}
+    nodeos  apply_context.cpp:124 exec_one
+```
+
+### Use require_auth2()
```

Comment:
Let's leave this out of the basics guide

Reply:
the way I see it all these three authorization checks are basic checks. and it is very good for the user to know about them from the get go.

Reply:
👍